src/classic/clvm_tools/stages/stage_2/optimize.rs

```diff
@@ -670,19 +729,26 @@ pub fn optimize_sexp_<'a>(
                 }
 
                 if equal_to(allocator, start_r, r) {
-                    return Ok(r);
+                    let memo_rc: &RefCell<HashMap<String, NodePtr>> = memo.borrow();
+                    memo_rc.replace_with(|mr| {
+                        let mut work = HashMap::new();
+                        swap(&mut work, mr);
```

Comment:
Why not mutate `mr` directly, and return it?

Reply:
you can't move out of just a mutable reference and replace_with returns the same type.  There's probably a more idiomatic way to do this, but that's what i landed on.

Reply:
structurally i like the enclosure of the mutation (if things need thread synchronization at some later time it'll remain the same)